packages/nextjs-mf/src/plugins/container/InvertedContainerRuntimeModule.ts

```diff
@@ -1,139 +1,69 @@
+    if (!this.compilation || !this.chunk || !this.compilation.chunkGraph) {
       return '';
     }
 
-    const { name, debug } = this.options;
-    const containerEntryModule = this.resolveContainerModule() as
-      | (Module & { _name: string })
+    const { name } = this.options;
+    const containerEntryModule = this.findEntryModuleOfContainer() as
+      | Module
       | undefined;
-    const containerName = containerEntryModule?._name || name;
-    const chunk = this.chunk;
 
-    const containerModuleId =
-      containerEntryModule?.id || containerEntryModule?.debugId;
-    // const hasEnsurechunkHandlers = !this.compilation.runtimeRequirements.has(RuntimeGlobals.ensureChunkHandlers)) {
-    const isPartialContainer = chunk.runtime === this.options.runtime;
-    if (!(containerName && containerModuleId) || !isPartialContainer) {
+    const containerModuleId = containerEntryModule
+      ? this.compilation.chunkGraph.getModuleId(containerEntryModule)
+      : false;
+
+    if (!containerModuleId) {
       return '';
     }
-    const globalObject = `globalThis.__remote_scope__`;
-    const containerScope = `${RuntimeGlobals.global}`;
-    const sharedObjectString = this.generateSharedObjectString();
+
+    const containerModuleIdJSON = JSON.stringify(containerModuleId);
+    const nameJSON = JSON.stringify(name);
 
     return Template.asString([
-      'var innerRemote;',
-      Template.indent([
-        'function attachRemote (resolve) {',
-        `  innerRemote = __webpack_require__(${JSON.stringify(
-          containerModuleId,
-        )});`,
-        `  if(${globalObject} && !${globalObject}[${JSON.stringify(name)}]) {`,
-        `    ${globalObject}[${JSON.stringify(name)}] = innerRemote;`,
-        `  } else if(${containerScope} && !${containerScope}[${JSON.stringify(
-          name,
-        )}]) {`,
-        `    ${containerScope}[${JSON.stringify(name)}] = innerRemote;`,
-        '  }',
-        `  __webpack_require__.S.default = new Proxy({${sharedObjectString}}`,
-        ' , {',
-        '    get: function(target, property) {',
-        '      return target[property];',
-        '    },',
-        '    set: function(target, property, value) {',
-        '      target[property] = value;',
-        '      return true;',
-        '    }',
-        '  });',
-        '  if(resolve) resolve(innerRemote);',
-        '}',
-      ]),
-      `if(!(${globalObject} && ${globalObject}[${JSON.stringify(
-        name,
-      )}]) && !(${containerScope} && ${containerScope}[${JSON.stringify(
-        name,
-      )}])) {`,
+      `var innerRemote;`,
+      `function attachRemote (resolve) {`,
       Template.indent([
-        '  if (__webpack_require__.O) {',
-        `  __webpack_require__.O(0, ["${this.chunk.id}"], function() {`,
-        '    attachRemote();',
-        '  }, 0);',
-        '} else {',
-        'attachRemote();',
-        '}',
+        `if(__webpack_require__.m[${containerModuleIdJSON}]) {`,
```

Comment:
since containerPlugin knows .federation runtime entry. We should be able to either make new chunk and require.ensure or add eager module to force that module to embed in all runtime chunks, like it does in remoteEntry automatically.